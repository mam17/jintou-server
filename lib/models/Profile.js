"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.profileSchema = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _mongoose = _interopRequireDefault(require("mongoose"));

var _validator = _interopRequireDefault(require("validator"));

var _momentTimezone = _interopRequireDefault(require("moment-timezone"));

var _constants = require("../constants");

var _errorMessages = _interopRequireDefault(require("../constants/errorMessages"));

/* eslint-disable no-underscore-dangle */
var authorizedUserAge = function authorizedUserAge() {
  return (0, _momentTimezone["default"])(Date.now(), 'America/Sao_Paulo').subtract(_constants.AUTHORIZED_USER_AGE, 'year').add(1, 'day').toString();
};

var profileSchema = _mongoose["default"].Schema({
  _id: _mongoose["default"].Schema.Types.ObjectId,
  name: {
    type: String,
    required: true,
    trim: true
  },
  birthday: {
    type: Date,
    required: true,
    validate: function validate(value) {
      if (_validator["default"].isAfter(value.toString(), authorizedUserAge())) {
        throw new Error(_errorMessages["default"].USER_AGE_NOT_AUTHORIZED);
      }
    }
  },
  images: [{
    _id: _mongoose["default"].Schema.Types.ObjectId,
    image: {
      type: String,
      required: false
    }
  }],
  loc: {
    type: {
      type: String
    },
    coordinates: []
  },
  birthplace: {
    placeId: {
      type: String,
      required: true
    },
    description: {
      type: String,
      required: true
    },
    lat: {
      type: String,
      required: true
    },
    lng: {
      type: String,
      required: true
    }
  },
  genre: {
    type: String,
    "enum": ['WOMAN', 'MAN', 'HUMAN'],
    required: true
  },
  sexualOrientations: [{
    type: String,
    "enum": ['HETERO', 'GAY', 'LESBIAN', 'BISEXUAL', 'ASEXUAL', 'DEMISEXUAL', 'PANSEXUAL', 'QUEER', 'QUESTIONING', 'OTHER'],
    required: true
  }],
  astral: {
    type: _mongoose["default"].Schema.Types.ObjectId,
    ref: 'Astral'
  },
  eyes: {
    type: String,
    required: true
  },
  graduation: {
    "class": {
      type: String,
      required: true
    },
    description: {
      type: String,
      required: true
    },
    placeId: {
      type: String,
      required: true
    }
  },
  occupation: {
    type: String,
    required: true
  },
  residence: {
    description: {
      type: String,
      required: true
    },
    placeId: {
      type: String,
      required: true
    }
  }
});

exports.profileSchema = profileSchema;
profileSchema.index({
  loc: '2dsphere'
});

profileSchema.methods.addProfileImage = /*#__PURE__*/function () {
  var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(imageUrl) {
    var image;
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            image = this.images[0];
            this.images = [{
              _id: image._id,
              image: imageUrl
            }];
            _context.next = 4;
            return this.save();

          case 4:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));

  return function (_x) {
    return _ref.apply(this, arguments);
  };
}();

var Profile = _mongoose["default"].model('Profile', profileSchema);

var _default = Profile;
exports["default"] = _default;