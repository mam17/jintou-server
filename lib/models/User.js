"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _mongoose = _interopRequireDefault(require("mongoose"));

var _validator = _interopRequireDefault(require("validator"));

/* eslint-disable no-underscore-dangle */
var userSchema = _mongoose["default"].Schema({
  _id: _mongoose["default"].Schema.Types.ObjectId,
  email: {
    type: String,
    required: true,
    unique: true,
    lowercase: true,
    validate: function validate(value) {
      if (!_validator["default"].isEmail(value)) {
        throw new Error({
          error: 'Invalid Email address'
        });
      }
    }
  },
  password: {
    type: String,
    required: true,
    minLength: 7
  },
  token: {
    type: String,
    required: true
  },
  profileStatus: {
    type: String,
    "enum": ['PENDING', 'CREATION', 'COMPLETED'],
    required: true,
    "default": 'PENDING'
  },
  configs: {
    maxDistance: {
      type: Number,
      required: true,
      "default": 100
    },
    love: {
      range: [Number],
      genre: {
        type: String,
        "enum": ['WOMAN', 'MAN', 'ALL'],
        required: false
      }
    },
    friendShip: {
      range: [Number],
      genre: {
        type: String,
        "enum": ['WOMAN', 'MAN', 'ALL'],
        required: false
      }
    }
  }
});

var User = _mongoose["default"].model('User', userSchema);

var _default = User;
exports["default"] = _default;