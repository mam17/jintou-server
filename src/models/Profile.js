/* eslint-disable no-underscore-dangle */
import mongoose from 'mongoose';
import validator from 'validator';
import moment from 'moment-timezone';
import { AUTHORIZED_USER_AGE } from '../constants';
import ERROR_MESSAGES from '../constants/errorMessages';
import { TextTypes } from './Astral';

const authorizedUserAge = () =>
  moment(Date.now(), 'America/Sao_Paulo')
    .subtract(AUTHORIZED_USER_AGE, 'year')
    .add(1, 'day')
    .toString();

export const profileSchema = mongoose.Schema({
  _id: mongoose.Schema.Types.ObjectId,
  name: {
    type: String,
    required: true,
    trim: true
  },
  birthday: {
    type: Date,
    required: true,
    validate: value => {
      if (validator.isAfter(value.toString(), authorizedUserAge())) {
        throw new Error(ERROR_MESSAGES.USER_AGE_NOT_AUTHORIZED);
      }
    }
  },
  images: [
    {
      _id: mongoose.Schema.Types.ObjectId,
      image: {
        type: String,
        required: false
      }
    }
  ],
  loc: {
    type: {
      type: String
    },
    coordinates: []
  },
  birthplace: {
    placeId: {
      type: String,
      required: true
    },
    description: {
      type: String,
      required: true
    },
    lat: {
      type: String,
      required: true
    },
    lng: {
      type: String,
      required: true
    }
  },
  genre: {
    type: String,
    enum: ['WOMAN', 'MAN', 'HUMAN'],
    required: true
  },
  sexualOrientations: [
    {
      type: String,
      enum: [
        'HETERO',
        'GAY',
        'LESBIAN',
        'BISEXUAL',
        'ASEXUAL',
        'DEMISEXUAL',
        'PANSEXUAL',
        'QUEER',
        'QUESTIONING',
        'OTHER'
      ],
      required: true
    }
  ],
  astral: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Astral'
  },
  eyes: {
    type: String,
    required: true
  },
  graduation: {
    class: {
      type: String,
      required: true
    },
    description: {
      type: String,
      required: true
    },
    placeId: {
      type: String,
      required: true
    }
  },
  occupation: {
    type: String,
    required: true
  },
  residence: {
    description: {
      type: String,
      required: true
    },
    placeId: {
      type: String,
      required: true
    }
  },
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  shownTexts: [
    {
      type: String,
      enum: TextTypes,
      default: [...TextTypes],
      required: true
    }
  ]
});

profileSchema.index({ loc: '2dsphere' });

profileSchema.methods.addProfileImage = async function(imageUrl) {
  const image = this.images[0];

  this.images = [{ _id: image._id, image: imageUrl }];
  await this.save();
};

const Profile = mongoose.model('Profile', profileSchema);

export default Profile;
