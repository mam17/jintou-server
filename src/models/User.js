/* eslint-disable no-underscore-dangle */
import mongoose from 'mongoose';
import validator from 'validator';
import { TextTypes } from './Astral';

const userSchema = mongoose.Schema({
  _id: mongoose.Schema.Types.ObjectId,
  email: {
    type: String,
    required: true,
    unique: true,
    lowercase: true,
    validate: value => {
      if (!validator.isEmail(value)) {
        throw new Error({ error: 'Invalid Email address' });
      }
    }
  },
  password: {
    type: String,
    required: true,
    minLength: 7
  },
  token: {
    type: String,
    required: true
  },
  profileStatus: {
    type: String,
    enum: ['PENDING', 'CREATION', 'COMPLETED'],
    required: true,
    default: 'PENDING'
  },
  configs: {
    maxDistance: {
      type: Number,
      required: true,
      default: 100
    },
    love: {
      range: [Number],
      genre: {
        type: String,
        enum: ['WOMAN', 'MAN', 'ALL'],
        required: false
      }
    },
    friendShip: {
      range: [Number],
      genre: {
        type: String,
        enum: ['WOMAN', 'MAN', 'ALL'],
        required: false
      }
    }
  },
  plan: {
    type: String,
    enum: TextTypes,
    default: 'MERCURIO',
    required: true
  }
});

const User = mongoose.model('User', userSchema);

export default User;
