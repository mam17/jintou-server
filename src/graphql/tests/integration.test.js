import axios from 'axios';
import moment from 'moment';
import { createTestClient } from 'apollo-server-integration-testing';
import mongoose from 'mongoose';
import generarateName from '../../util/randomName';
import FILE_B64 from '../../__mocks__/fileb64';
import { typeDefs, resolvers, context, ApolloServer, app } from '../..';
import Profile from '../../models/Profile';
import User from '../../models/User';

const CREATE_PROFILE = `
  mutation($file: String, $name: String, $birthday: String, $input: BirthplaceInput) {
    createProfile(file: $file, name: $name, birthday: $birthday, input: $input) {
      _id
      name
      birthday
      images {
        _id
        image
      }
      birthplace {
        placeId
        description
        lat
        lng
      }
      loc {
        coordinates
      }
    }
  }
`;

// const max = 99999999;
// const min = 9999999;
// const getRandomNumber = () => Math.floor(Math.random() * (max - min + 1)) + min;
// const getPostData = () => ({
//   name: generarateName(),
//   email: `${getRandomNumber()}@gmail.com`,
//   password: getRandomNumber()
// });

test('Creating random user and profile', async () => {
  jest.setTimeout(30000);

  const PORT = 3000;

  const mockedUser = {
    name: 'Server Testing',
    email: 'server-testing@gmail.com',
    password: '123TEST123'
  };

  const birthplace = {
    placeId: '123a123b123c',
    description: generarateName(),
    lat: '-16.6782694',
    lng: '-49.3233247'
  };

  const apolloServer = new ApolloServer({
    typeDefs,
    resolvers,
    dataSources: () => ({}),
    context
  });

  await Profile.findOneAndRemove({ email: mockedUser.email });
  await User.findOneAndRemove({ email: mockedUser.email });

  const server = app.listen(3000);

  await new Promise(resolve => server.on('listening', () => resolve()));

  await axios({
    method: 'post',
    url: `http://localhost:${PORT}/users`,
    data: {
      ...mockedUser
    }
  });

  const {
    data: { token },
    data: user
  } = await axios({
    method: 'post',
    url: `http://localhost:${PORT}/users/login`,
    data: {
      ...mockedUser
    }
  });

  const userKeys = Object.keys(user);

  const newProfile = {
    name: user.name,
    birthday: moment(new Date())
      .subtract(19, 'years')
      .toString(),
    file: FILE_B64,
    input: {
      ...birthplace
    }
  };

  const { mutate } = createTestClient({
    apolloServer,
    extendMockRequest: {
      headers: {
        Authorization: `Bearer ${token}`
      }
    }
  });

  const {
    data: { createProfile: profile }
  } = await mutate(CREATE_PROFILE, {
    variables: { ...newProfile }
  });

  const profileKeys = Object.keys(profile).reduce((accumulator, key) => {
    if (Array.isArray(profile[key])) {
      return [...accumulator, ...Object.keys(profile[key][0])];
    }
    if (typeof profile[key] === 'object') {
      return [...accumulator, ...Object.keys(profile[key])];
    }
    return [...accumulator, key];
  }, []);

  await new Promise((resolve, reject) =>
    server.close(err => {
      if (err) {
        return reject();
      }
      return resolve();
    })
  );

  await mongoose.disconnect();

  expect(token).not.toBeNull();
  expect(profileKeys).toMatchSnapshot();
  expect(userKeys).toMatchSnapshot();
});
