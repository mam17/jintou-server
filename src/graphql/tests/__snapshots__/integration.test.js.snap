// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating random user and profile 1`] = `
Array [
  "_id",
  "name",
  "birthday",
  "_id",
  "image",
  "placeId",
  "description",
  "lat",
  "lng",
  "coordinates",
]
`;

exports[`Creating random user and profile 2`] = `
Array [
  "_id",
  "name",
  "email",
  "token",
  "hasProfile",
]
`;
